circup.py

```diff
@@ -642,7 +650,11 @@ def freeze():  # pragma: no cover
     """
     logger.info("Freeze")
     modules = find_modules()
-    if modules:
+    if JSON_OUT:
+        modules = [ (m.row[0], m.row[2]) for m in modules]
```

Comment:
Am I correct in guessing that a list of lists is the data structure that best suits VSCode..?

If not, why not just dump a JSON object? Such a dictionary like object feels (to me) like a more natural way to express key/value pairs than a list of lists. I'm also painfully aware that I'm probably missing something here (hence the VSCode question above).

:slightly_smiling_face:

Reply:
Honestly, the only reason I did it this way is that it looked the most like the data I was already scraping from the output. I'm going to rework the whole thing in the morning so it looks more like a complete API, which includes adding json output for install and update, which I totally skipped because on the vscode side I'm just displaying the output directly. Might as well do it right :D